src/app_task.cpp

```diff
@@ -356,3 +386,67 @@ void AppTask::DispatchEvent(const AppEvent &event)
 		LOG_INF("Event received with no handler. Dropping event.");
 	}
 }
+
+void AppTask::UpdateTemperatureClusterState(const AppEvent &)
+{
+	static constexpr int16_t kTemperatureMeasurementAttributeMaxValue = 0x7fff;
+	static constexpr int16_t kTemperatureMeasurementAttributeMinValue = 0x954d;
+	static constexpr int16_t kTemperatureMeasurementAttributeInvalidValue = 0x8000;
+	static constexpr uint8_t kTemperatureMeasurementEndpointId = 1;
+
+	struct sensor_value sTemperature;
+	EmberAfStatus status;
+	int err = sensor_sample_fetch(sTemperatureSensor);
+	if (!err) {
+		err = sensor_channel_get(sTemperatureSensor, SENSOR_CHAN_DIE_TEMP, &sTemperature);
+		if (!err) {
+			int16_t newValue = static_cast<int16_t>(sTemperature.val1 * 100 +
+								sTemperature.val2 / 10000);
+
+			if (newValue > kTemperatureMeasurementAttributeMaxValue ||
+			    newValue < kTemperatureMeasurementAttributeMinValue) {
+				/* Read value exceeds permitted limits, so assign invalid value code to it. */
+				newValue = kTemperatureMeasurementAttributeInvalidValue;
```

Comment:
I think the attribute is nullable so we can set it to null instead. AFAIK there's no such thing as invalid value in this cluster.